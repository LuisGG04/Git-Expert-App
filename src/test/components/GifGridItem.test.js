import "@testing-library/jest-dom"
import React from "react"
import { shallow } from "enzyme"
import {GifGridItem} from "../../components/GifGridItem"

describe('Testeo en el componente GifGridItem', () => {

    const title = "Hola Mundo"
    const url = "https://localhost/"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    
    test('Debe de mostrar correctamente al componente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe mostrar un parrafo con texto', () => {
        const p = wrapper.find("p")
        expect(p.text().trim()).toBe(title)
    })

    test('Debe de mostrar la imagen igual a url y alt', () => {
        const img = wrapper.find("img")
        expect(img.prop("src")).toBe(url)
        expect(img.prop("alt")).toBe(title)
    })

    test('Debe de tener la clase "animate__fadeIn"', () => {
        
        const div = wrapper.find("div")
        const className = div.prop("className")
        expect(className.includes("animate__fadeIn")).toBe(true)
    })
    
})

