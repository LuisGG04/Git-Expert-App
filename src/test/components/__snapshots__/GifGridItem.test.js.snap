// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo en el componente GifGridItem Debe de mostrar correctamente al componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola Mundo"
    src="https://localhost/"
  />
  <p>
     
    Hola Mundo
     
  </p>
</div>
`;
